ext/standard/array.c

```diff
@@ -2057,13 +2055,25 @@ PHP_FUNCTION(array_fill_keys)
 }
 /* }}} */
 
-#define RANGE_CHECK_INIT_ARRAY(start, end) do { \
+#define RANGE_CHECK_DOUBLE_INIT_ARRAY(start, end) do { \
 		double __calc_size = ((start - end) / step) + 1; \
-		if (fabs(__calc_size) >= (double)HT_MAX_SIZE) { \
-			php_error_docref(NULL, E_WARNING, "The supplied range exceeds the maximum array size: start=%0.0f end=%0.0f", start > end ? end : start, start > end ? start : end); \
+		if (__calc_size > (double)HT_MAX_SIZE) { \
```

Comment:
Ok, with argument swapping it will work as well, that's correct then.

With HT_MAX_SIZE - yes, like in the place i've linked earlier, or fe here http://lxr.php.net/xref/PHP_7_0/Zend/zend_hash.c#zend_hash_check_size . Or you can check other usages. Size has to be always < HT_MAX_SIZE, thus the previuos >=.

Thanks.